content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,295 @@
+
+The input must have a minimum contrast ratio of 3:1 to pass WCAG AA-compliance. This is defined in [WCAG success criterion 1.4.11](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html).
+
+### Caption
+
+A caption may be used to provide additional context about the field to help users fill in the correct data or explain how the data will be used. Caption text should be as short as possible.
+
+Caption text may be displayed with a validation message, or it may be hidden if it is redundant information.
+
+Caption text may be used to augment the [label](#label), but should not be redundant with the label or any other parts of the form control. If the caption feels redundant, try removing it. 
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065190-5ef9d8db-c9ca-4aec-a2ea-321a81cc080e.png" />
+    <Caption>Use caption text that shows new information and provides helpful context</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065192-f1fd1c28-2031-4bb1-836b-607d14a3d4f4.png" />
+    <Caption>Don’t use caption text that is redundant</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Validation message
+
+A single validation message may be displayed to provide helpful information for a user to complete their task. For example: explain an invalid value and submit the form.
+
+An individual checkbox or radio should not have its own validation.
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065196-b0146ade-eccb-4a71-b1b6-f4a82b96c115.png" />
+    <Caption>Show validation for the group of inputs</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065199-6cfa4893-55cb-4f81-a7d6-4b6ff66339eb.png" />
+    <Caption>Don’t show validation for each input</Caption>
+  </Dont>
+</DoDontContainer>
+
+Information from the caption should not be repeated in the error message. Show the validation message and remove the caption. A validation message makes the field easier to spot when a user is scanning for invalid fields.
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065196-b0146ade-eccb-4a71-b1b6-f4a82b96c115.png" />
+    <Caption>Hide a caption that is redundant with the validation message</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065194-8f123836-5cda-4eea-8f0f-f1c016aa197c.png" />
+    <Caption>Don’t show a repetitive validation message</Caption>
+  </Dont>
+</DoDontContainer>
+
+For more information about form validation, see the [validation guidelines](#invalid-input).
+
+## Input methods
```

Comment:
Shouldn't this section be structured as a list of components for easier parsing, instead of prose with static examples?

[Example](https://primer-6d38263007-26441320.drafts.github.io/components/action-list#application):

> ![Screenshot of Application section in Action list interface guidelines](https://user-images.githubusercontent.com/293280/166301732-1a450074-3321-4398-8aa5-3d9604d14b8a.png)

Reply:
I really like this. Thanks for the example!